lib/components/guess.js

```diff
@@ -21,10 +21,20 @@ export default class Guess extends React.Component{
     this.setState({ attempt: e.target.value });
   }
 
+  setMin(e) {
+    this.setState({ min: e.target.value })
+  }
+
+  setMax(e) {
+    this.setState({ max: e.target.value })
+  }
+
   updateRange(e) {
-    this.setState({ min: e.target.value });
-    this.setState({ max: e.target.value });
-    generate();
+    let newMin = parseInt(this.state.min);
+    let newMax = parseInt(this.state.max);
+    this.setState({ min: newMin });
+    this.setState({ max: newMax });
+    this.generate();
```

Comment:
The issue seems to be that you are acting on old state. In React, the state is only set once you have "left" the function in which you're changing it. So, you are applying new state, but when you call `generate()` - in which you are relying on that new state - the state hasn't been changed yet. 

This can be solved by adding a callback to the `setState` function, which will execute once your state is changed. 
It would look something like this: `this.setState({ min: newMin, max: newMax}, () => this.generate() )`. You might have to bind the `this` context since we're using an arrow function.